Tests relying on Testcontainers no longer work with the recent version of Docker

Some of the current tests don't work with the recent version of Docker (29.0.0+) as follows. Upgrading Testcontainers 2.0.2+ solves this issue.

```
$ docker version
Client: Docker Engine - Community
 Version:           29.1.1
 API version:       1.52
...

Server: Docker Engine - Community
 Engine:
  Version:          29.1.1
  API version:      1.52 (minimum version 1.44)
...

$ mvn clean install -DskipTests
$ mvn scalatest:test -pl spark/common/pom.xml -Dsuites=org.apache.sedona.sql.OsmReaderTest

(snip)

[INFO] --- scalatest:2.2.0:test (default-cli) @ sedona-spark-common-3.4_2.12 ---
[INFO] ScalaTest report directory: /home/sekikn/repos/apache/GeoSpark/spark/common/target/surefire-reports
Run starting. Expected test count is: 7
OsmReaderTest:
Loading OSM data
- should load OSM data from local file
- should be able to process planet osm files
- should parse normal nodes
- should parse dense nodes
- should be able to read from osm file on s3 *** FAILED ***
  java.lang.IllegalStateException: Could not find a valid Docker environment. Please see logs and check configuration
  at org.testcontainers.dockerclient.DockerClientProviderStrategy.lambda$getFirstValidStrategy$7(DockerClientProviderStrategy.java:277)
  at java.base/java.util.Optional.orElseThrow(Optional.java:403)
  at org.testcontainers.dockerclient.DockerClientProviderStrategy.getFirstValidStrategy(DockerClientProviderStrategy.java:268)
  at org.testcontainers.DockerClientFactory.getOrInitializeStrategy(DockerClientFactory.java:152)
  at org.testcontainers.DockerClientFactory.client(DockerClientFactory.java:194)
  at org.testcontainers.DockerClientFactory$1.getDockerClient(DockerClientFactory.java:106)
  at com.github.dockerjava.api.DockerClientDelegate.authConfig(DockerClientDelegate.java:109)
  at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:329)
  at org.apache.sedona.sql.OsmReaderTest.$anonfun$new$6(OsmReaderTest.scala:115)
  at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
  ...
- should properly assign values
- should not lose precision due to float to double conversion
Run completed in 6 seconds, 200 milliseconds.
Total number of tests run: 7
Suites: completed 1, aborted 0
Tests: succeeded 6, failed 1, canceled 0, ignored 0, pending 0
*** 1 TEST FAILED ***
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  8.463 s
[INFO] Finished at: 2025-12-03T20:51:28+09:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.scalatest:scalatest-maven-plugin:2.2.0:test (default-cli) on project sedona-spark-common-3.4_2.12: There are test failures -> [Help 1]
```

--- a/spark/common/pom.xml
+++ b/spark/common/pom.xml
@@ -250,14 +250,14 @@
         </dependency>
         <dependency>
             <groupId>org.testcontainers</groupId>
-            <artifactId>minio</artifactId>
-            <version>1.20.0</version>
+            <artifactId>testcontainers-minio</artifactId>
+            <version>2.0.2</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.testcontainers</groupId>
-            <artifactId>mysql</artifactId>
-            <version>1.20.0</version>
+            <artifactId>testcontainers-mysql</artifactId>
+            <version>2.0.2</version>
             <scope>test</scope>
         </dependency>
         <dependency>

--- a/spark/spark-3.4/pom.xml
+++ b/spark/spark-3.4/pom.xml
@@ -135,13 +135,13 @@
         <dependency>
             <groupId>org.testcontainers</groupId>
             <artifactId>testcontainers</artifactId>
-            <version>1.20.1</version>
+            <version>2.0.2</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.testcontainers</groupId>
-            <artifactId>minio</artifactId>
-            <version>1.20.0</version>
+            <artifactId>testcontainers-minio</artifactId>
+            <version>2.0.2</version>
             <scope>test</scope>
         </dependency>
         <dependency>

--- a/spark/spark-3.5/pom.xml
+++ b/spark/spark-3.5/pom.xml
@@ -135,13 +135,13 @@
         <dependency>
             <groupId>org.testcontainers</groupId>
             <artifactId>testcontainers</artifactId>
-            <version>1.20.1</version>
+            <version>2.0.2</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.testcontainers</groupId>
-            <artifactId>minio</artifactId>
-            <version>1.20.0</version>
+            <artifactId>testcontainers-minio</artifactId>
+            <version>2.0.2</version>
             <scope>test</scope>
         </dependency>
         <dependency>

--- a/spark/spark-4.0/pom.xml
+++ b/spark/spark-4.0/pom.xml
@@ -135,13 +135,13 @@
         <dependency>
             <groupId>org.testcontainers</groupId>
             <artifactId>testcontainers</artifactId>
-            <version>1.20.1</version>
+            <version>2.0.2</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.testcontainers</groupId>
-            <artifactId>minio</artifactId>
-            <version>1.20.0</version>
+            <artifactId>testcontainers-minio</artifactId>
+            <version>2.0.2</version>
             <scope>test</scope>
         </dependency>
         <dependency>

